clmm/cosmology/numcosmo.py

```diff
@@ -147,28 +147,30 @@ def _eval_linear_matter_powerspectrum(self, k_vals, redshift):
 
         # Using the EH transfer function as this is the 
         # default for the CCL backend as well
-        ps = Nc.PowspecMLTransfer.new (Nc.TransferFuncEH.new()) 
+        ps = Nc.PowspecMLTransfer.new(Nc.TransferFuncEH.new())
+        ps.set_kmin(1.0e-5)
+        ps.set_kmax(1.0)
 
         # Instead, computing the PS from the CLASS backend of Numcosmo
         # ps  = Nc.PowspecMLCBE.new ()
         # ps.peek_cbe().props.use_ppf = True
 
         if self.be_cosmo.reion is None:
-            reion = Nc.HIReionCamb.new ()
-            self.be_cosmo.add_submodel (reion)
+            reion = Nc.HIReionCamb.new()
+            self.be_cosmo.add_submodel(reion)
         if self.be_cosmo.prim is None:
-            prim  = Nc.HIPrimPowerLaw.new ()
-            self.be_cosmo.add_submodel (prim)
+            prim  = Nc.HIPrimPowerLaw.new()
+            self.be_cosmo.add_submodel(prim)
             # The default CLMM cosmology has ns=0.96 and sigma8=0.8
             # Need to adapt the NC cosmology accordingly
             self.be_cosmo.prim.props.n_SA = 0.96
-            psf = Ncm.PowspecFilter.new (ps, Ncm.PowspecFilterType.TOPHAT)
-            old_amplitude = np.exp (self.be_cosmo.prim.props.ln10e10ASA)
-            self.be_cosmo.prim.props.ln10e10ASA = np.log ((0.8 / self.be_cosmo.sigma8(psf))**2 * old_amplitude)
+            sigma8 = ps.sigma_tophat_R(self.be_cosmo, 1.0e-11, 0.0, 8.0/self.be_cosmo.h())
```

Comment:
I suspect that $10^{-11}$ is too much, I don't know your target precision but I imagine that $10^{-7}$ or $10^{-8}$ should be enough.

Reply:
yeah, 1e-8 is probably good enough.